qiskit_finance/applications/optimization/portfolio_optimization.py

```diff
@@ -103,6 +117,35 @@ def interpret(self, result: Union[OptimizationResult, np.ndarray]) -> List[int]:
         x = self._result_to_x(result)
         return [i for i, x_i in enumerate(x) if x_i]
 
+    def _check_compatibility(self) -> None:
+        """Check the compatibility of given variables"""
+        if len(self._expected_returns) != len(self._covariances) or not all(
+            len(self._expected_returns) == len(row) for row in self._covariances
+        ):
+            raise QiskitFinanceError(
+                "The sizes of expected_returns and covariances do not match. ",
+                f"expected_returns: {self._expected_returns}, covariances: {self._covariances}.",
+            )
+        if self._bounds is not None:
+            if (
+                not isinstance(self._bounds, list)
+                or not all(isinstance(lb_, int) for lb_, _ in self._bounds)
+                or not all(isinstance(ub_, int) for _, ub_ in self._bounds)
+            ):
+                raise QiskitFinanceError(
+                    f"The bounds must be a list of tuples of integers. {self._bounds}",
+                )
+            if any(ub_ < lb_ for lb_, ub_ in self._bounds):
+                raise QiskitFinanceError(
+                    "The upper bound of each variable must be larger than the lower bound of that.",
```

Comment:
```suggestion
                    "The upper bound of each variable in the list must be larger than its lower bound.",
```
Just a suggestion that may make it a little easier to read. 

I have a question too. Do we not support the case where the upper and lower bound are the same - maybe that fixes the value so it should be taken out of the optimization problem.

Reply:
Thank you! I fixed the message. 

The current implementation supports the case where the upper and lower bound are the same. It raises an error only when the upper bound of some variable is lower than its lower bound.

Reply:
Yes in looking at the code that is true - the error message states though that it "must be larger than.." which would not be true if they were equal and that is supported. Wwould it therefore more correct to say "must be greater than or equal to the lower bound" in the error message.

Reply:
Ah, you're right. the error message is inappropriate. I fixed it. Thx!